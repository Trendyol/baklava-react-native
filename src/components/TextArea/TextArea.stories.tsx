import React, { useState } from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Text from '../Text/Text';
import TextArea from './TextArea';
import Box from '../Box/Box';
import Button from '../Button/Button';

const sizeList = ['small', 'medium', 'large'];

const TextAreaMeta: ComponentMeta<typeof TextArea> = {
  title: 'TextArea',
  component: TextArea,
  argTypes: {
    size: {
      options: sizeList,
      control: { type: 'radios' },
    },
  },
  args: {
    size: 'medium',
    label: 'Label',
    placeholder: 'Type something...',
    labelFixed: false,
    helpText: 'Helper Text',
    error: false,
    disabled: false,
    errorText: 'Error Text',
  },
};

export default TextAreaMeta;

type TextAreaStory = ComponentStory<typeof TextArea>;

export const Basic: TextAreaStory = args => (
  <Box>
    <Box px="2xs">
      <Text py="2xs" variant="subtitle01Bold">
        Text Area
      </Text>
    </Box>

    <TextArea {...args} />
  </Box>
);

export const Cases: TextAreaStory = args => {
  const [state, setState] = useState(args);

  return (
    <Box>
      <Box px="2xs">
        <Text py="2xs" variant="subtitle01Bold">
          Text Area Cases
        </Text>
      </Box>

      <TextArea {...args} disabled={true} />

      <Box pt="2xl">
        <TextArea {...state} />

        <Box flexDirection="row" px="2xs" marginLeft="2xs" marginTop="xs">
          <Button
            flex={1}
            kind="danger"
            variant="secondary"
            size="s"
            onPress={() => setState({ ...state, error: true, errorText: null })}
            label="Toggle Error"
            filled={true}
            mr="2xs"
          />
          <Button
            flex={1}
            size="s"
            kind="danger"
            variant="secondary"
            onPress={() =>
              setState({
                ...state,
                error: true,
                errorText: 'Error text',
              })
            }
            label="Error with Text"
            filled={true}
            mr="2xs"
          />
        </Box>
      </Box>
    </Box>
  );
};

export const Sizes: TextAreaStory = () => (
  <Box>
    <Box px="2xs">
      <Text py="2xs" variant="subtitle01Bold">
        Text Area Sizes
      </Text>
    </Box>

    <Box py="xs">
      <TextArea
        size="large"
        label="Label"
        placeholder="Large"
        helpText="Help Text"
      />
    </Box>

    <Box py="xs">
      <TextArea
        size="medium"
        label="Label"
        placeholder="Medium"
        helpText="Help Text"
      />
    </Box>

    <Box py="xs">
      <TextArea
        size="small"
        label="Label"
        placeholder="Small"
        helpText="Help Text"
      />
    </Box>
  </Box>
);

export const Types: TextAreaStory = args => {
  return (
    <Box>
      <Box px="2xs">
        <Text py="2xs" variant="subtitle01Bold">
          Text Area Types
        </Text>
      </Box>

      <TextArea {...args} labelFixed={true} />

      <TextArea {...args} />

      <TextArea {...args} label={null} />
    </Box>
  );
};
