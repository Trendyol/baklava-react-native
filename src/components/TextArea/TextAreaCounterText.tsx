import React from 'react';
import Box from '../Box/Box';
import Text from '../Text/Text';
import { getCounterText, getTextColor } from './utils';

export const TextAreaCounterText = React.memo(
  ({
    counterText,
    errorState,
  }: {
    counterText?: number;
    errorState: boolean;
  }): JSX.Element | null => {
    const textColor = getTextColor({ errorState });

    const content = getCounterText({
      counterText,
      errorState,
    });

    return (
      <Box mt="3xs" ml="m">
        <Text variant="subtitle04Regular" color={textColor}>
          {`${content}/200`}
        </Text>
      </Box>
    );
  },
);
