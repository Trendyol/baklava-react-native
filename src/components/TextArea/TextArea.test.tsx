import React from 'react';
import { act } from 'react-test-renderer';
import { fireEvent, render } from '../../test-utils';
import theme from '../../theme';
import TextArea, { TextInputHandles } from './TextArea';

describe('TextArea', () => {
  test('should render TextArea correctly', () => {
    // when
    const { toJSON } = render(<TextArea label="label" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render label fixed TextArea correctly', () => {
    // when
    const { toJSON } = render(
      <TextArea label="label" labelFixed placeholder="placeholder" />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render disabled TextArea correctly', () => {
    // when
    const { toJSON } = render(<TextArea disabled label="label" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render TextArea with help text correctly', () => {
    // when
    const { toJSON } = render(<TextArea label="label" helpText="helpText" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render TextArea with error state correctly', () => {
    // when
    const { toJSON } = render(<TextArea label="label" error />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render TextArea with error state and error text correctly', () => {
    // when
    const { toJSON } = render(
      <TextArea label="label" error errorText="errorText" />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render TextArea with small size correctly', () => {
    // when
    const { toJSON } = render(<TextArea label="label" size="small" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render TextArea with medium size correctly', () => {
    // when
    const { toJSON } = render(<TextArea label="label" size="medium" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render TextArea with large size correctly', () => {
    // when
    const { toJSON } = render(<TextArea label="label" size="large" />);

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should focus TextArea correctly', () => {
    // when
    const { getByTestId } = render(<TextArea label="label" />);

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.neutralLighter,
    );

    // when
    act(() => {
      fireEvent(getByTestId('textArea'), 'onFocus');
    });

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.primaryKey,
    );
  });

  test('should not focus and blur TextArea when TextArea is disabled', () => {
    // when
    const { getByTestId } = render(<TextArea label="label" disabled />);

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.neutralLighter,
    );

    // when
    act(() => {
      const textArea = getByTestId('textArea');
      fireEvent(textArea, 'onFocus');
      fireEvent(textArea, 'onBlur');
    });

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.neutralLighter,
    );
  });

  test('should blur TextArea without value correctly', () => {
    // when
    const { getByTestId } = render(<TextArea label="label" />);

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.neutralLighter,
    );

    // when
    act(() => {
      const textArea = getByTestId('textArea');
      fireEvent(textArea, 'onFocus');
      fireEvent(textArea, 'onBlur');
    });

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.neutralLighter,
    );
  });

  test('should blur TextArea with value correctly', () => {
    // when
    const { getByTestId } = render(<TextArea label="label" value="value" />);

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.neutralLighter,
    );

    // when
    act(() => {
      const textArea = getByTestId('textArea');
      fireEvent(textArea, 'onFocus');
      fireEvent(textArea, 'onBlur');
    });

    // then
    expect(getByTestId('textArea-box').props.style[0].borderColor).toBe(
      theme.colors.neutralLighter,
    );
  });

  test('should change text of TextArea without value', () => {
    // when
    const { getByTestId } = render(<TextArea label="label" />);
    const textArea = getByTestId('textArea');
    fireEvent.changeText(textArea, 'e-mail@mail.com');

    // then
    expect(textArea.props.value).toBe('e-mail@mail.com');
  });

  test('should change text of textArea with value', () => {
    // when
    const { getByTestId } = render(<TextArea label="label" value="value" />);
    const textArea = getByTestId('textArea');
    fireEvent.changeText(textArea, 'e-mail@mail.com');

    // then
    expect(textArea.props.value).toBe('value');
  });

  test('should not change text of TextArea when TextArea is disabled', () => {
    // when
    const { getByTestId } = render(<TextArea label="label" disabled />);
    const textArea = getByTestId('textArea');
    fireEvent.changeText(textArea, 'e-mail@mail.com');

    // then
    expect(textArea.props.value).not.toBe('e-mail@mail.com');
  });

  test('should change text of TextArea when TextArea has defaultValue', () => {
    // when
    const { getByTestId } = render(
      <TextArea label="label" defaultValue="defaultValue" />,
    );
    const textArea = getByTestId('textArea');
    fireEvent.changeText(textArea, 'e-mail@mail.com');

    // then
    expect(textArea.props.value).toBe('e-mail@mail.com');
  });

  test('should access textAreas handle functions correctly', () => {
    // given
    const ref = React.createRef<TextInputHandles>();

    // when
    render(<TextArea label="label" ref={ref} />);

    act(() => {
      ref.current?.focus();
      ref.current?.blur();
      ref.current?.clear();
      ref.current?.isFocused();
      ref.current?.setNativeProps({ text: '' });
    });

    // then
    expect(ref.current?.focus).toBeTruthy();
  });
});
