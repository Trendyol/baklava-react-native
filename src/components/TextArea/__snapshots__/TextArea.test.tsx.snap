// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render TextArea correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 94,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#95A1B5",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#AFBBCA",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 88,
            "paddingTop": 12,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render TextArea with error state and error text correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 112,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "red",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#FF5043",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 88,
            "paddingTop": 12,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#FF5043",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          errorText
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#FF5043",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render TextArea with error state correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 94,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "red",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#FF5043",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 88,
            "paddingTop": 12,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#FF5043",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render TextArea with help text correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 112,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#95A1B5",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#AFBBCA",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 88,
            "paddingTop": 12,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          helpText
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render TextArea with large size correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 110,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#95A1B5",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#AFBBCA",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 104,
            "paddingTop": 16,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render TextArea with medium size correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 94,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#95A1B5",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#AFBBCA",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 88,
            "paddingTop": 12,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render TextArea with small size correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 86,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "color": "#95A1B5",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#AFBBCA",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 80,
            "paddingTop": 8,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render disabled TextArea correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="none"
    style={
      Object {
        "bottom": 0,
        "height": 94,
        "left": 12,
        "position": "absolute",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingLeft": 16,
          },
        ]
      }
      testID="outlined-label-box"
    >
      <Text
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#95A1B5",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "height": 16,
            "lineHeight": 16,
            "paddingLeft": 4,
            "paddingRight": 4,
            "top": 8,
          }
        }
      >
        label
      </Text>
    </View>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F1F2F7",
          "borderColor": "#AFBBCA",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={true}
      editable={false}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 88,
            "paddingTop": 12,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`TextArea should render label fixed TextArea correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginBottom": 4,
        },
      ]
    }
    testID="fixed-label-box"
  >
    <Text
      accessibilityLabel="fixed-label"
      accessible={true}
      style={
        Array [
          Object {
            "color": "#6E7787",
            "fontFamily": "Rubik-Medium",
            "fontSize": 12,
            "fontWeight": "500",
            "lineHeight": 14,
          },
        ]
      }
      testID="fixed-label"
    >
      label
    </Text>
  </View>
  <View
    accessibilityLabel="textArea-box"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#AFBBCA",
          "borderRadius": 6,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "zIndex": 0,
        },
      ]
    }
    testID="textArea-box"
  >
    <TextInput
      accessibilityLabel="textArea"
      bg="transparent"
      disabled={false}
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="placeholder"
      placeholderTextColor="#95A1B5"
      style={
        Array [
          Object {
            "flex": 1,
            "height": 88,
            "paddingTop": 12,
          },
        ]
      }
      testID="textArea"
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-start",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "marginLeft": 16,
              "marginTop": 4,
            },
          ]
        }
      >
        <Text
          accessible={false}
          style={
            Array [
              Object {
                "color": "#95A1B5",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "fontWeight": "400",
                "lineHeight": 14,
              },
            ]
          }
        >
          0/200
        </Text>
      </View>
    </View>
  </View>
</View>
`;
